PROTOCOL.md

```diff
@@ -265,19 +265,25 @@ The following is an example `metaData` action:
 
-The path of a file acts as the primary key for the entry in the set of files.
-When an `add` action is encountered for a path that is already present in the table, statistics and other information from the latest version should replace that from any previous version.
-As such, additional statistics can be added for a path already present in the table by adding it again.
+Every _logical file_ of the table (referred to as just a _file_ going forward) is represented by a path to a data file, combined with a Deletion Vector (DV) that indicates which rows of the data file are no longer in the table. The path of the data file acts as the primary key for the entry in the set of files. Deletion Vectors are an optional feature. In tables that do not have this feature enabled, all Deletion Vectors are considered to be empty.
+
+When an `add` action is encountered for a file (i.e., a data file and Deletion Vector combination) that is already present in the table, statistics and other information from the latest version should replace that from any previous version.
+The primary key for the entry of a file in the set of files is a tuple of the data file's `path` as well as a unique id describing the DV.
 
 The `remove` action includes a timestamp that indicates when the removal occurred.
-Physical deletion of the file can happen lazily after some user-specified expiration time threshold.
+Physical deletion of the data and Deletion Vector file can happen lazily after some user-specified expiration time threshold.
 This delay allows concurrent readers to continue to execute against a stale snapshot of the data.
 A `remove` action should remain in the state of the table as a _tombstone_ until it has expired.
-A tombstone expires when the creation timestamp of the delta file exceeds the expiration threshold added to the `remove` action timestamp.
+A tombstone expires when *current time* (according to the node performing the cleanup) exceeds the expiration threshold added to the `remove` action timestamp.
+
+In the following statements, a "DV" can refer to either a specific Deletion Vector, or to the empty deletion vector that indicates that no rows are deleted. Since actions within a given Delta commit are not guaranteed to be applied in order, a **valid** version is restricted to contain at most one file action for any one combination of data file and DV. Moreover, for simplicity it is required that there is at most one file action for any data file (regardless of DV).
+That means specifically that for any commit…
 
-Since actions within a given Delta file are not guaranteed to be applied in order, it is not valid for multiple file operations with the same path to exist in a single version.
+ - it is **legal** for the same data file to occur in an `add` action and a `remove` action, but with two different DVs (one of which may be the empty DV).
+ - it is **legal** for the same data file to be added and/or removed and also occur in an `AddCDCFile` action.
```

Comment:
You cannot talk about `AddCDCFile` here since that has not been introduced.

Reply:
Actually this [PR](https://github.com/delta-io/delta/pull/1300) needs to be merged first for the CDC specifications.